src/main/java/org/springframework/data/couchbase/repository/CouchbaseRepository.java

```diff
@@ -32,15 +48,135 @@
 @NoRepositoryBean
 public interface CouchbaseRepository<T, ID> extends PagingAndSortingRepository<T, ID> {
 
+	// all the inherited methods need to be supplemented by ones that take Options, CollectionName and ScopeName args
+
+	@Override
+	Optional<T> findById(ID id);
+	Optional<T> findById(ID id, QueryOptions... options);
```

Comment:
That is exactly what we want to avoid. The repository must not accept store-specific types as we do not want the calling code to interact with store specifics.

Store specifics at the Template API level are fine.